### PR TITLE
spike on how to generalize landing page

### PR DESCRIPTION
**Pull request by amaliwanag2**
This is a proof of concept of how to generalize the landing page
@kevin please see PR annotations
Screen Shot 2021-09-19 at 2 30 27 PM
Gatsby autopopulates the fields you can query in igraphql and i thought that was cool

### CODE REVIEW

File: gatsby-config.esm.js [hunk 1]
```diff
@@ -46,6 +46,13 @@ module.exports = {
         excludes: ['/help', '/contact'],
       },
     },
+    `gatsby-transformer-json`,
```
Comment (amaliwanag2) [inline gatsby-config.esm.js:49]: to ingest json files into our data layer

File: gatsby-node.esm.js [hunk 2]
```diff
@@ -74,6 +74,84 @@ exports.createPages = async ({ actions, graphql }) => {
...
+  v4CauseSpecificHomepages.allCauseSpecificJsonDataJson.edges.forEach(
+    ({ node: { path, data } }) => {
+      createPage({
+        path: `${path}/`,
+        component: catsLandingPage, // this will be new component that takes all data as props,
+        context: {
+          data,
+        },
+      })
+      response.data.allReferrersYaml.edges.forEach(({ node }) => {
```
Comment (amaliwanag2) [inline gatsby-node.esm.js:137]: @kmjennison do we need a referrer for each version of v4?

File: src/data/causeSpecificJSONData/cats.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+  "path": "cats",
```
Comment (amaliwanag2) [inline src/data/causeSpecificJSONData/cats.json:2]: mapped this to what we have in the mocks, I know it's subject to change

File: src/data/causeSpecificJSONData/cats.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        "img2": "TFACIntro2.png",
+        "img2Subtext": "Earn Money for cat training programs.",
+        "img3": "TFACIntro3.png",
+        "img3Subtext": "Help get more cats adopted."
+      },
+      "Mission": {
+        "title": "Our Mission",
+        "subtitle": "Help shelter cats get adopted",
+        "text": "We have partnered with The Greater Good Charities' The Jackson Galaxy Project to provide treats to shelter cats for FREE! How can we do this? Through ad revenue! Every time you open a new tab, there will be two small ads in the right corner of your home screen. Companies pay for this space to advertise on our platform, and from there we can put that money to good use by giving it back to an organization doing great work for our furry friends. Tab for Cats is absolutely free for our users, and a great way to make an impact without breaking the bank."
+      }
```
Comment (amaliwanag2) [inline src/data/causeSpecificJSONData/cats.json:42]: I'm still missing 3 sections, but you get the idea

File: src/data/causeSpecificJSONData/cats.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        "secondary": "29BEBA",
+        "secondaryShadow": "15AAA6",
+        "background": "FAEFE1"
+      }
+    },
+    "sections": {
+      "landing": {
+        "title": "Open tabs.  Help shelter cats!",
+        "subtitle": "Every tab opened raises money for cats in need of an extra, helping paw. It’s so easy a cat could do it.",
+        "ctaImg": "cat1.png"
```
Comment (amaliwanag2) [inline src/data/causeSpecificJSONData/cats.json:16]: I'm thinking we can just pass down the file name of the images, and then we can lazy load them inside the react component.  Cursory googling makes it seem like this is very doable

File: gatsby-node.esm.js [hunk 6]
```diff
@@ -74,6 +74,84 @@ exports.createPages = async ({ actions, graphql }) => {
...
+        path
+      }
+    }
+  }
+  `)
+  v4CauseSpecificHomepages.allCauseSpecificJsonDataJson.edges.forEach(
+    ({ node: { path, data } }) => {
+      createPage({
+        path: `${path}/`,
+        component: catsLandingPage, // this will be new component that takes all data as props,
```
Comment (amaliwanag2) [inline gatsby-node.esm.js:132]: component would be new generalized component that reads all data from props and is purely presentational
Doc is unclear if this context prop is populated on first render, so not sure if we'll need to make a loading component

File: gatsby-node.esm.js [hunk 7]
```diff
@@ -74,6 +74,84 @@ exports.createPages = async ({ actions, graphql }) => {
       },
     })
   })
+  const v4CauseSpecificHomepages = await graphql(`
+  allCauseSpecificJsonDataJson(limit: 1000) {
```
Comment (amaliwanag2) [inline gatsby-node.esm.js:78]: Igraphql UI build this query for me which is really convenient :)

File: gatsby-node.esm.js (hunk 2 above)
Comment (kmjennison) [reply gatsby-node.esm.js:137]: Yes, we should.

File: src/data/causeSpecificJSONData/cats.json [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    "styles": {
+      "colors": {
+        "primary": "FF6A08",
+        "secondary": "29BEBA",
+        "secondaryShadow": "15AAA6",
+        "background": "FAEFE1"
+      }
+    },
+    "metadata": {
+      "title": "Tab for Cats - Home",
```
Comment (amaliwanag2) [inline src/data/causeSpecificJSONData/cats.json:13]: @kmjennison am i missing anything in the metadata section?  This was based on what I saw in cats

Comment (kmjennison) [reply src/data/causeSpecificJSONData/cats.json:13]: That's all I can think of for now. I know this is a draft, but just some feedback:
* Assuming `causeSpecificKeywords` is joined with a generic list of keywords, then don't need "charity"
* Probably don't need `titleTemplate`. Don't think we'll customize it per page?
* Typo: `cannonicalUrl` --> `canonicalUrl`. Though, I'd suggest we add a `pagePath` key that's used in Gatsby page generation and use it for canonical URL (don't want them to get out of sync).

File: src/data/causeSpecificJSONData/cats.json [hunk 9]
```diff
@@ -0,0 +1,55 @@
+{
```
Comment (kmjennison) [inline src/data/causeSpecificJSONData/cats.json:1]: Name directory something like `causes`.

File: src/data/causeSpecificJSONData/cats.json [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+  "path": "cats",
+  "data": {
+    "styles": {
+      "colors": {
+        "primary": "FF6A08",
```
Comment (kmjennison) [inline src/data/causeSpecificJSONData/cats.json:6]: Include `#` in hex values. It's less logic for the page and should increase flexibility to use rgba.

File: src/data/causeSpecificJSONData/cats.json [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        "title": "Open tabs.  Help shelter cats!",
+        "subtitle": "Every tab opened raises money for cats in need of an extra, helping paw. It’s so easy a cat could do it.",
+        "ctaImg": "cat1.png"
+      },
+      "moneyRaised": {
+        "moneyImg": "catsMoneyRaised.png"
+      },
+      "charityIntro": {
+        "title": "Being a shelter cat is hard.",
+        "subTitle": "Our impact is everything to us. And, we’re so proud of the impact we’ve been able to drive, together, thus far.",
```
Comment (kmjennison) [inline src/data/causeSpecificJSONData/cats.json:32]: "subTitle" --> "subtitle"

File: src/data/causeSpecificJSONData/cats.json [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        "introImg2": "introImg2.png",
+        "introImg2Subtext": "More cats adopted",
+        "link": "https://greatergood.org/jackson-galaxy"
+      },
+      "TFACIntro": {
+        "title": "The easiest way to raise money.",
+        "img1": "TFACIntro1.png",
+        "img1Subtext": "Open new tabs on your browser.",
+        "img2": "TFACIntro2.png",
+        "img2Subtext": "Earn Money for cat training programs.",
```
Comment (kmjennison) [inline src/data/causeSpecificJSONData/cats.json:44]: "Money" --> "money"

File: src/data/causeSpecificJSONData/cats.json [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+        "img3": "TFACIntro3.png",
+        "img3Subtext": "Help get more cats adopted."
+      },
+      "Mission": {
```
Comment (kmjennison) [inline src/data/causeSpecificJSONData/cats.json:48]: lowercase "mission"

File: src/data/causeSpecificJSONData/cats.json (hunk 4 above)
Comment (kmjennison) [reply src/data/causeSpecificJSONData/cats.json:52]: Still need these sections for the spec, right?

File: gatsby-node.esm.js (hunk 2 above)
Comment (kmjennison) [reply gatsby-node.esm.js:137]: @amaliwanag2 Did this make it into your spec tickets?

**Review by kmjennison — CHANGES_REQUESTED** [review]

File: gatsby-config.esm.js [hunk 14]
```diff
@@ -46,6 +53,10 @@ module.exports = {
...
+    'gatsby-plugin-image',
+    'gatsby-plugin-sharp',
```
Comment (amaliwanag2) [inline gatsby-config.esm.js:58]: there's a lot of image optimization that i didn't do that plugin-sharp can do.  Mostly just transforming the images to match the pixels they'll use at build time

File: gatsby-node.esm.js [hunk 15]
```diff
@@ -74,6 +74,91 @@ exports.createPages = async ({ actions, graphql }) => {
...
+                  introImg2Subtext
+                  subTitle
+                  title
+                }
+                landing {
+                  subtitle
+                  title
+                  ctaImg {
+                    childImageSharp {
+                      gatsbyImageData(width: 200, formats: AUTO)
```
Comment (amaliwanag2) [inline gatsby-node.esm.js:111]: we'll want to move away from formats auto and add more things here.  Can do this in the query builder tool

File: package.json [hunk 16]
```diff
@@ -18,14 +20,19 @@
     "dotenv-extended": "^2.9.0",
     "gatsby": "^3.14.0",
     "gatsby-plugin-facebook-pixel": "^1.0.5",
+    "gatsby-plugin-image": "^1.14.0",
     "gatsby-plugin-material-ui": "^4.0.3",
+    "gatsby-plugin-mdx": "^2.14.0",
```
Comment (amaliwanag2) [inline package.json:25]: mdx is not necessary as we're not going to be using it

File: src/data/markdownpages/secondPage.mdx [hunk 17]
```diff
@@ -0,0 +1,67 @@
+## Copyblog/yet-another-post.mdx: copy code to clipboard
```
Comment (amaliwanag2) [inline src/data/markdownpages/secondPage.mdx:1]: i'll eventually close out this pr but just commiting this so I can come back to it if we switch to mdx at some point

**Pull request closed without merging**